pkg/asset/machines/worker.go

```diff
@@ -173,6 +174,7 @@ func (w *Worker) Dependencies() []asset.Asset {
 		// it is put in the dependencies but not fetched in Generate
 		&installconfig.PlatformCredsCheck{},
 		&installconfig.InstallConfig{},
+		&kubeconfig.AdminClient{},
```

Comment:
This is going to have a side-effect of creating the kube certs during manifest generation. I don't know that that is **necessarily** a problem. But if I recall correctly, we've tried to delay the cert creation until the ignition step so that the user does not accidentally use expired or soon-to-be expired certs from a `create manifests` done previously.

Reply:
+1.

I am also in agreement with you and wary about the admin cert/key in the machineconfigs. This seems like using a sledgehammer on a tiny nail not to mention super insecure.

I was wondering if there is an alternative we should pursue. Basically the MCO outputs two different rendered configs. Once by the Machine config server (mcs), which includes contents of /etc/kubernetes/kubeconfig and the other is the machineconfig object which omits this info. Since we cannot access the mcs (blocked on purpose within the pod network), we wanted to rely on the machineconfig objects, which is not complete.

Reply:
I think I'm going to rework this to send the kubeconfig as a secret to the operator-machine-api namespace.